src/App.tsx

```diff
@@ -1,43 +1,294 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import { client } from './utils/fetchClient';
+import { TodoList } from './components/TodoList';
+import {
+  Error, ErrorTimeout, ErrorButtonHandler, ErrorCloser,
+} from './types/Error';
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [needUpdate, setNeedUpdate] = useState(false);
+
+  const [formInput, setFormInput] = useState('');
+
+  const [selectedTodo, setSelectedTodo] = useState('All');
+
+  const [addTodoError, setAddTodoError] = useState(false);
+  const [deleteTodoError, setDeleteTodoError] = useState(false);
+  const [updateTodoError, setUpdateTodoError] = useState(false);
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const updateTodos = async () => {
+    await client.get<Todo[]>(`/todos?userId=${user?.id}`)
+      .then(setTodosFromServer);
+  };
+
+  const allTodos = () => {
+    setTodos(todosFromServer);
+    setSelectedTodo('All');
+  };
+
+  const activeTodos = () => {
+    setTodos(todosFromServer.filter(todo => todo.completed === false));
+    setSelectedTodo('Active');
+  };
+
+  const completedTodos = () => {
+    setTodos(todosFromServer.filter(todo => todo.completed === true));
+    setSelectedTodo('Completed');
+  };
+
+  const allCompletedButtonHandler = async () => {
+    const date = {
+      completed: true,
+    };
+
+    todosFromServer.filter(todo => todo.completed === false)
+      .forEach(async todo => {
+        await client.patch(`/todos/${todo.id}`, date);
+      });
+
+    setNeedUpdate(true);
+  };
+
+  const buttonErrorHandler = (callback: ErrorButtonHandler) => {
+    callback(false);
+  };
+
+  const errorCloser = (close: ErrorCloser) => {
+    switch (close) {
```

Comment:
why this variable names `close`. It's inlogical `close === 'updatetodo'`. 
Rename it for more meaningful variable